containers/unit_tests/TestWithoutInitializing.hpp

```diff
@@ -465,28 +490,58 @@ TEST(TEST_CATEGORY, create_mirror_no_init_dynrankview) {
 TEST(TEST_CATEGORY, create_mirror_no_init_dynrankview_viewctor) {
   using namespace Kokkos::Test::Tools;
   listen_tool_events(Config::DisableAll(), Config::EnableKernels());
-  Kokkos::DynRankView<int, Kokkos::DefaultExecutionSpace> device_view(
-      "device view", 10);
+  Kokkos::DynRankView<int, TEST_EXECSPACE> device_view("device view", 10);
   Kokkos::DynRankView<int, Kokkos::HostSpace> host_view("host view", 10);
+  auto device_memory_space = typename TEST_EXECSPACE::memory_space{};
 
   auto success = validate_absence(
       [&]() {
         auto mirror_device = Kokkos::create_mirror(
             Kokkos::view_alloc(Kokkos::WithoutInitializing), device_view);
         ASSERT_EQ(device_view.size(), mirror_device.size());
+#ifndef KOKKOS_ENABLE_CUDA_UVM
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device)::memory_space,
+                           Kokkos::HostSpace>);
+#else
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device)::memory_space,
+                           decltype(device_memory_space)>);
+#endif
         auto mirror_host = Kokkos::create_mirror(
-            Kokkos::view_alloc(Kokkos::WithoutInitializing,
-                               Kokkos::DefaultExecutionSpace{}),
+            Kokkos::view_alloc(Kokkos::WithoutInitializing, TEST_EXECSPACE{},
+                               device_memory_space),
             host_view);
         ASSERT_EQ(host_view.size(), mirror_host.size());
+    // FIXME_MSVC
+#ifndef KOKKOS_COMPILER_MSVC
+        static_assert(
+            std::is_same_v<typename decltype(mirror_host)::memory_space,
+                           decltype(device_memory_space)>);
+#endif
```

Comment:
I don't understand why these `static_assert`s fail with `MSVC`. They work fine on `Serial` only builds with other compilers.
I would propose to keep this as-is for now unless someone can help me debug this with MSVC.

Reply:
Fixed it.